request_headers.go

```diff
@@ -0,0 +1,115 @@
+package abstractions
+
+import "strings"
+
+//RequestHeaders represents a collection of request headers
+type RequestHeaders struct {
+	headers map[string]map[string]bool
+}
+
+//NewRequestHeaders creates a new RequestHeaders
+func NewRequestHeaders() *RequestHeaders {
+	return &RequestHeaders{
+		headers: make(map[string]map[string]bool),
+	}
+}
+
+func (r *RequestHeaders) normalizeHeaderKey(key string) string {
+	return strings.ToLower(key)
+}
+
+//Add adds a new header or append a new value to an existing header
+func (r *RequestHeaders) Add(key string, value string, additionalValues ...string) {
+	if key == "" || value == "" {
```

Comment:
Empty string key value " " will be accepted

Reply:
so the suggestion here is to call the normalize key and then check?

Reply:
yes

Reply:
thanks, done.